eng/native/configurecompiler.cmake

```diff
@@ -166,8 +155,7 @@ if (CLR_CMAKE_ENABLE_SANITIZERS)
       # /RTC1 is added by default by CMake and incompatible with ASAN, so remove it.
       string(REPLACE "/RTC1" "" CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG}")
       string(REPLACE "/RTC1" "" CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG}")
-      string(REPLACE "/RTC1" "" CMAKE_SHARED_LINKER_FLAGS_DEBUG "${CMAKE_SHARED_LINKER_FLAGS_DEBUG}")
-      string(REPLACE "/RTC1" "" CMAKE_EXE_LINKER_FLAGS_DEBUG "${CMAKE_EXE_LINKER_FLAGS_DEBUG}")
+      string(REPLACE "/RTC1" $<IF:$<CONFIG:DEBUG>,"","/RTC1"> LINK_OPTIONS "${LINK_OPTIONS}")
```

Comment:
I never thought about using a generator expression in a replacement here. Interesting! However, we don't need to keep `/RTC1` on all platforms (it's only added-by-default in DEBUG). Also, this option is set specifically in the old-style flags. Let's leave the changes to this flag as-is.